root/etc/e-smith/templates/etc/ssh/sshd_config/70Restricted2Sftp

```diff
@@ -0,0 +1,27 @@
+{
+    # restrict groups to sftp
+
+    use strict;
+    use esmith::ConfigDB;
+    my $db = esmith::ConfigDB->open_ro();
+
+    my @SftpRestricted = ();
+    foreach ( split(',',$db->get_prop('sshd','AllowGroups') || '')) {
+        my ($group, $sftp) = split(':', $_);
+        if(!$group || !$sftp) {
+            next;
+        }
+        push @SftpRestricted, $group if ($sftp eq 'SftpRestricted');
+    }
+
+    if (@SftpRestricted) {
+        @SftpRestricted = join (',',@SftpRestricted);
+        #Spaces are not allowed in group name => we quote
+        $OUT .= "Match Group \"@SftpRestricted\"\n";
+        $OUT .= "    ForceCommand internal-sftp\n";
+        $OUT .= "Match All\n";
+    }
+
+
+    $OUT .= "";
```

Comment:
What does it mean? :)

```suggestion
```

Reply:
when the array is empty it ends with a `zero` in a template I found a more elegant way